elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -264,6 +266,20 @@ export abstract class BaseTransaction {
 			errors.push(idError);
 		}
 
+		/* tslint:disable */
+		if (this.type !== (<typeof BaseTransaction>this.constructor).TYPE) {
```

Comment:
Shouldn't this check be done in the constructor?

Reply:
No